server/src/infra/configuration.rs

```diff
@@ -81,6 +81,26 @@ impl std::default::Default for LdapsOptions {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, derive_builder::Builder)]
+#[builder(pattern = "owned")]
+pub struct PluginOptions {
+    #[builder(default = "true")]
+    pub enabled: bool,
+    #[builder(default = "false")]
+    pub allow_on_password_update: bool,
+    #[builder(default = r#"None"#)]
+    pub kvscope: Option<String>,
```

Comment:
I know you talked about it, but I still don't see why we need this scope. Why _isn't_ this always the plugin name?

Reply:
I think it's valuable for the user to be able to override what the plugin implementer has declared.

It would allow use cases like:

- Clearing the existing state at deployment, by adding an index to the name. Effectively providing a clean scope very time, without needing to do deletions on the DB.
- Debug if issues with a new version of a plugin is due to the existing state, by again giving a clean scope, without actually deleting the old.
- Maybe it's meaningful for some plugins to instantiate them more than once, configure them differently, and separate their state.
- Maybe it could be convenient to give a group of plugins the same scope.

I just think it's a convenient control for the user to have.

Reply:
SGTM